ext/liquid_c/parser.c

```diff
@@ -119,12 +119,10 @@ static VALUE parse_variable(parser_t *p)
         }
     }
 
-    if (RARRAY_LEN(lookups) == 0 && TYPE(name) == T_STRING) {
-        if (rstring_eq(name, "nil") || rstring_eq(name, "null")) return Qnil;
-        if (rstring_eq(name, "true")) return Qtrue;
-        if (rstring_eq(name, "false")) return Qfalse;
-        if (rstring_eq(name, "blank")) return symBlank;
-        if (rstring_eq(name, "empty")) return symEmpty;
+    if (RARRAY_LEN(lookups) == 0) {
+        VALUE undefined = FIXNUM_P(-1);
```

Comment:
There was a patch submitted in 2002 to change this haha. http://blade.nagaokaut.ac.jp/cgi-bin/scat.rb/ruby/ruby-core/86